### PR TITLE
Reporting

### PR DESCRIPTION
**Pull request by VesnaT**
OWReport: New report widget.
Add report to most widgets.

### CODE REVIEW

**Comment by janezd** [comment]
@astaric , I assigned this to you now. Please check and merge (at your earliest convenience, this one is difficult to rebase).

File: Orange/canvas/report/report.py [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+        :param table: table to be reported
+        :type table:
+            QAbstractItemModel or QStandardItemModel or two-dimensional list or
+            any object with method `model()` that returns one of the above
+        :param header_rows: the number of rows that are marked as header rows
+        :type header_rows: int
+        :param header_columns:
+            the number of columns that are marked as header columns
+        :type header_columns: int
+        :param num_format: numeric format, e.g. `{:.3}`
```
Comment (kernc) [inline Orange/canvas/report/report.py]: :+1: for Napoleon here as well.

**Comment by kernc** [comment]
> This branch has conflicts that must be resolved

Could you rebase this and/or squash together relevant commits. :smile:

**Comment by janezd** [comment]
@VesnaT, rebase works OK (4x conflict in widget.py + 1x report.py, otherwise smooth) if I set `git config --global core.trustctime false` **before** I start rebasing. Alternatively, rebase with `git -c core.trustctime=false rebase master`. See http://stackoverflow.com/questions/5074136/git-rebase-fails-your-local-changes-to-the-following-files-would-be-overwritte. If you don't do this, you'll get the same weird stuff as the last time.

@kernc I'll use Napoleon when writing new docstrings.

**Pull request merged**